go/cmd/config/hasura_handler.go

```diff
@@ -162,6 +163,29 @@ func (h *HasuraHandler) handler(w http.ResponseWriter, r *http.Request) {
 		path := fmt.Sprintf("/api/v1/rules/%s/%s", request.Input.Namespace, request.Input.RuleGroupName)
 		httpresp, err := h.cortexQuery(request.Input.TenantID, "DELETE", path, "")
 		response = actions.ToDeleteResponse("Rule group", httpresp, err)
+
+	case "validateTenantName":
+		var request actions.ValidateTenantNamePayload
+		err = json.Unmarshal(reqbody, &request)
+		if err != nil {
+			writeGraphQLError(w, "invalid request payload")
+			return
+		}
+		regex := "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
```

Comment:
Compare with https://github.com/opstrace/opstrace/blob/main/packages/config/src/clusterconfig.ts#L33

Reply:
With the recent compability issues @nickbp encountered we're moved to only allowing `alphanumeric` characters in Tenant names, the regex you've highlighted allows `-` in them which needs to be removed. For reference I've updated and then more recently [simplified the UI validation regex](https://github.com/opstrace/opstrace/pull/999/files#diff-7325966d61eaee7624a540e5774552cdf48b41e5bbd72d76a9db24a29ea48062R23) to being:

```
/^[a-z0-9]{1,63}$/
```

Also there is a character limit of 63 as K8s and/or DNS doesn't like anything longer than that.

Reply:
In my understanding @jgehrcke introduced the RegEx above, because it's the very same K8S is using.
https://github.com/opstrace/opstrace/blob/d8194a3e9b9dedc5faaa7bce9d5b8fd45a96ba32/packages/config/src/clusterconfig.ts#L31